### ISSUE TITLE
Query parameter for readout of focal length, image distance, object distance

### ISSUE THREAD

**Issue by ariel-phet** [phetsims/geometric-optics#334]
In testing for #328 I noticed that attempting to get good values to plug into the image equation was a bit challenging. You have to be VERY careful and precise with the placement of the rulers, and in many situations the ruler is not the right length to measure the image distance in one measurement (so I was using the vertical ruler as a sign post -- first make a measurement from the optic, place the vertical ruler at the end of the horizontal ruler, and then make a second measurement to the image). For example:
![measurement](https://user-images.githubusercontent.com/4051174/155853588-4d2de2d4-c751-4d11-acf1-14368042478f.PNG)

It might be nice to consider having a query parameter that just had "read out values" so if people wanted precise values for the imaging equation they would be available. I know this sim is more of a virtual lab in the sense of measurements, so I am not advocating for this sort of readout to be a main feature of the sim, but if it is easy to add a query parameter for such a display it would be helpful (I also think it would be handy for checking the physics for QA or such in the future).

Something simple like the following with just values floating in the upper right would do:

![measurementquery](https://user-images.githubusercontent.com/4051174/155853802-e089d1d0-49ad-4345-9cc3-4801da0714c8.png)

Anyhow, just a suggestion, not at all a showstopper for publication.

**Comment by pixelzoom** [comment]
This would be easy to add as @ariel-phet described.   But it feels like a bit of a design hack to stick these values in like this. And I have many questions:

* If this is important, is a query parameter appropriate?  Should we instead add an "Image Equation" accordion box, roughy in the same location? 
* Is it sufficient to show just the f, do, di values, or should it also show the image equation?
* Should do and di actually be d<sub>o</sub> and d<sub>i</sub> respectively?
* Should we use PhET's standard math font for these symbols?

**Comment by ariel-phet** [comment]
@pixelzoom I have mixed feelings about adding this feature in to be a default part of the sim. I think the ethos of the design is for students to be able to generally discover this relationship and make measurements. The other questions you raise are probably worthy of a short design discussion if we decide to move forward with having some sort of display like this. Either way, there does seem to be a slight usability issue in terms of measure image or object distances that exceed the length of the ruler.

**Comment by ariel-phet** [comment]
@pixelzoom  -- if we are going there....

A tape measure (or allowing two tape measures) would frankly be ideal. I would think then you would have the tape measures be a bit "snappy" so that they preferentially snap to interesting points, like the center of the optic and the imaging points. A tape measure would also you to measure the diagonal if desired (which would allow further confirmation of the role of similar triangles). Our tape measures also have a readout so you could precisely measure distances. 

The one "advantage" of a ruler is that it allows some measurement error, so perhaps just not having a tape measure be "snappy" would also achieve that goal. These also seem like questions worthy of a short design discussion.

**Comment by pixelzoom** [comment]
2 tape measure is a toolbox is certainly doable.  It will come with a very large cost.  We spent an enormous amount of design & implementation time on the rulers (dynamic scale, alt input, hotkeys, iO,...) and we'll need to revist all of that for tape measures.

**Comment by ariel-phet** [comment]
@pixelzoom then my guess is that for the moment doing tape measures is not feasible and would perhaps be best as a separate issue as a design enhancement in future (might make a nice project for an intern/new dev/etc). We can keep this issue focused on whether we want to have the option for some sort of readout of values (which seems like it would be low cost comparatively).

**Comment by pixelzoom** [comment]
> ... We can keep this issue focused on whether we want to have the option for some sort of readout of values  ...

Yes, let's do that.  I'll move "rulers vs tape measure" to a new issue.

**Comment by pixelzoom** [comment]
@arouinfar and I discussed this yesterday.  @arouinfar is going to add comments about why we're not going to do this.

**Comment by pixelzoom** [comment]
3/3/2022 design meeting.

The consensus was that the best way to do this would be by adding dimensional arrows that should labels and values for important measurements.  We decided not to do this for the next release, defer and reconsider for a future release.

Notes:
* Add a "Values" checkbox
* Quantities: f, do, di (with correct signs), possibly arrow height
* For horizontal quantities, dimensional arrows, in the space above optic, drop leader lines to the optical axis.  They may need to be vertically staggered so that they don’t overlap when zoomed out.
* For vertical quantities,...?
* @ariel-phet suggested possibly showing values only for Arrow scene

**Comment by kathy-phet** [comment]
I would continue to defer this issue at this time to keep focused on current goal of a PhET-iO release, and see if it pops up as a user request and PhET-iO users use this sim in the future. (It might).
